src/modules/logicModule/abstracts/ERC20PaymentClientBase_v1.sol

```diff
@@ -86,12 +86,29 @@ abstract contract ERC20PaymentClientBase_v1 is
     /// @dev	The current cumulative amount of tokens outstanding.
     mapping(address => uint) internal _outstandingTokenAmounts;
 
+    /// @dev    The number of payment processor flags used by this payment
+    ///         client.
+    uint8 internal _flagCount;
+
+    /// @dev    The payment processor flags used by this payment client.
+    bytes32 internal _flags;
+
     /// @dev	Storage gap for future upgrades.
-    uint[50] private __gap;
+    uint[48] private __gap;
 
     //--------------------------------------------------------------------------
     // Internal Mutating Functions
 
+    /// @dev	Initializes the staking contract.
+    /// @param  flags_ The flags, represented as an array of uint8 containing
+    ///         the flag IDs between 0 and 255.
+    function __ERC20PaymentClientBase_v1_init(uint8[] memory flags_)
```

Comment:
So in your system you are now using 2 representations for setting the flags:

- flags are uint8[] type that is used (here in the setter function)
- flags are a bitmaps of type bytes32 (used everywhere elese, also in public getters adn events)

I'd just use a single represeentation (i.e. just use bitmaps)